enzy_htp/_interface/rosetta_interface.py

```diff
@@ -38,7 +46,1113 @@
+        pdb_files (List[str]): A list of paths to PDB files produced as part of the modeling results.
+        parent_job (ClusterJob): The cluster job associated with the RosettaScripts execution.
+    """
+    score_file: str
+    pdb_files: List[str]
+    parent_job: ClusterJob
+
+
+class RosettaScriptsEngine(ModelingEngine):
+    """Engine class for executing RosettaScripts protocols.
+
+    This class is responsible for managing the preparation and execution of RosettaScripts workflow,
+    defining input setup, job submission, and result parsing.
+
+    Attributes:
+        _parent_interface: The interface through which the engine communicates with external components.
+        _work_dir (str): The directory where input/output files for the job are stored.
+        _protocol (RosettaScriptsProtocol): The protocol defining how RosettaScripts should be executed.
+        _opts (RosettaOptions): Options to configure the Rosetta job.
+        _cluster_job_config (Dict): Configuration parameters for cluster job submission.
+    """
+
+    def __init__(self, interface, protocol: RosettaScriptsProtocol, opts: RosettaOptions, cluster_job_config: Dict, work_dir: str):
+        """Initializes a RosettaScriptsEngine instance.
+
+        Args:
+            interface: The interface for communication with external modules.
+            protocol (RosettaScriptsProtocol): The protocol to be executed.
+            opts (RosettaOptions): Options specific to the Rosetta execution.
+            cluster_job_config (Dict): Configuration settings for the cluster job.
+            work_dir (str): Path to the working directory for storing job files.
+        """
+        self._parent_interface = interface
+        self._work_dir = work_dir
+        self._protocol = protocol
+        self._opts = opts
+        self._cluster_job_config = cluster_job_config
+
+    @property
+    def parent_interface(self):
+        """Gets the parent interface for communication with external components.
+
+        Returns:
+            The parent interface.
+        """
+        return self._parent_interface
+
+    @property
+    def work_dir(self) -> str:
+        """Gets the working directory for storing input/output files.
+
+        Returns:
+            str: Path to the working directory.
+        """
+        return self._work_dir
+
+    @property
+    def protocol(self) -> RosettaScriptsProtocol:
+        """Gets the protocol associated with the engine.
+
+        Returns:
+            RosettaScriptsProtocol: The executing protocol.
+        """
+        return self._protocol
+
+    @property
+    def opts(self) -> RosettaOptions:
+        """Gets the options for the Rosetta job execution.
+
+        Returns:
+            RosettaOptions: The options set for the job.
+        """
+        return self._opts
+
+    @property
+    def cluster_job_config(self):
+        """Gets the configuration for cluster job submission.
+
+        Returns:
```

Comment:
These hardcoded paths for the ACCRE cluster should be moved to a configuration file or handled through the environment manager to make the code more portable.